src/servers/http_server_v2.cc

```diff
@@ -1254,11 +1254,34 @@ HTTPAPIServerV2::HandleSystemSharedMemory(
         size_t buffer_len = evbuffer_get_length(req->buffer_in);
         err = EVBufferToJson(&register_request, v, &v_idx, buffer_len, n);
         if (err == nullptr) {
-          const char* shm_key = register_request["key"].GetString();
-          uint64_t offset = register_request["offset"].GetInt();
-          uint64_t byte_size = register_request["byte_size"].GetInt();
-          err = shm_manager_->RegisterSystemSharedMemory(
-              region_name, shm_key, offset, byte_size);
+          const auto& key_itr = register_request.FindMember("key");
+          if (key_itr == register_request.MemberEnd()) {
+            err = TRITONSERVER_ErrorNew(
+                TRITONSERVER_ERROR_INVALID_ARG,
+                "Shared memory register request has no 'key' field");
+          } else {
+            const char* shm_key = key_itr->value.GetString();
+            uint64_t offset = 0, byte_size = 0;
+            const auto& offset_itr = register_request.FindMember("offset");
```

Comment:
> The “shared_memory_offset” parameter is optional and defaults to zero. The other two parameters are required. If only one of the two is given the server should return an appropriate error. 

From the specification. It looks like the shared_memory_offset should be optional and server should not throw the error in case offset is absent.

Reply:
Thanks for catching this. Fixed this